### PR TITLE
Revert "Change default cardinality to low for metrics (#7522)"

### PR DESCRIPTION
**Pull request by artursouza**
# Description

This reverts commit c3baea50a19e8e00793f5d79929edce45fc4c5e2.

Proposal to postpone decision to make low cardinality default, done in https://github.com/dapr/dapr/pull/7522

## Issue reference

Please reference the issue this PR will close: N/A

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: charts/dapr/crds/configuration.yaml [hunk 1]
```diff
@@ -253,8 +253,9 @@ spec:
                       the HTTP server
                     properties:
                       increasedCardinality:
-                        description: |-
-                          If false (the default), metrics for the HTTP server are collected with increased cardinality.
+                        description: 'If true, metrics for the HTTP server are collected
+                          with increased cardinality. The default is true in Dapr 1.13,
+                          but will be changed to false in 1.14+'
```
Comment (artursouza) [inline charts/dapr/crds/configuration.yaml:258]: I can update this message too, just waiting for this run to finish.

Comment (ItalyPaleAle) [inline charts/dapr/crds/configuration.yaml:258]: You may need to update this

File: pkg/apis/configuration/v1alpha1/types.go [hunk 2]
```diff
@@ -219,7 +219,9 @@ type MetricSpec struct {
 
 // MetricHTTP defines configuration for metrics for the HTTP server
 type MetricHTTP struct {
-	// If false (the default), metrics for the HTTP server are collected with increased cardinality.
+	// If false, metrics for the HTTP server are collected with increased cardinality.
+	// The default is true in Dapr 1.13, but will be changed to false in 1.14+
+	// TODO @ItalyPaleAle [MetricsCardinality] Change default in 1.14
```
Comment (ItalyPaleAle) [inline pkg/apis/configuration/v1alpha1/types.go:224]: Update this

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

**Comment by yaron2** [comment]
Context:

After testing the low cardinality mode as default for a while and having received a satisfactory amount of user feedback, it becomes clear that while this enabling or disabling by default cannot be summarized into a right/wrong decision and both carry pros and cons that will better optimize for a certain class of users. The overall feedback received is that users want the ability to easily turn off high cardinality metrics if they need to, and a subset of users would very much benefit more granular control over the metrics configuration, such as path matching and regex, to allow power users to tailor metrics to their needs. With the implications of making metrics less usable out of the box for new users and potentially disrupting dashboards, monitoring and observability platform ingesting Dapr metrics of existing users, it would be preferable to follow how way sister projects like Istio and Linkerd handle the situation, which is to keep high cardinality as default and provide users with the best experience of opting out or arriving at a desired custom configuration.

**Review by cicoyle — APPROVED** [review]
Changes look fine, but there are some failing int tests that I believe should be passing before this gets merged

**Comment by artursouza** [comment]
> Changes look fine, but there are some failing int tests that I believe should be passing before this gets merged

I think I fixed now. Let's see.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**